src/Moryx.AbstractionLayer.Resources.Endpoints/ResourceManagementController.cs

```diff
@@ -144,30 +145,48 @@ public ActionResult<Entry> InvokeMethod(long id, string method, Entry parameters
         public ActionResult<ResourceModel> ConstructWithParameters(string type, string method = null, [FromBody(EmptyBodyBehavior = EmptyBodyBehavior.Allow)] Entry arguments = null)
         {
             var trustedType = WebUtility.HtmlEncode(type);
-
-            return method == null ? Construct(trustedType, null)
-                : Construct(trustedType, new MethodEntry { Name = method, Parameters = arguments });
+            if (method is null)
+                return Construct(trustedType);
+            else
+                return Construct(trustedType, new MethodEntry { Name = method, Parameters = arguments });
         }
 
-        private ActionResult<ResourceModel> Construct(string type, MethodEntry method)
+        private ActionResult<ResourceModel> Construct(string type)
         {
-            var resource = (Resource)Activator.CreateInstance(_resourceTypeTree[type].ResourceType);
-            if (resource is null)
+            Resource resource;
+            try
+            {
+                resource = (Resource)Activator.CreateInstance(_resourceTypeTree[type].ResourceType);
+            }
+            catch (Exception)
+            {
                 return NotFound(new MoryxExceptionResponse { Title = Strings.RESOURCE_NOT_FOUND });
+            }
 
             ValueProviderExecutor.Execute(resource, new ValueProviderExecutorSettings()
                 .AddFilter(new DataMemberAttributeValueProviderFilter(false))
                 .AddDefaultValueProvider());
 
-            if (method != null)
-                EntryConvert.InvokeMethod(resource, method, _serialization);
-
             var model = new ResourceToModelConverter(_resourceTypeTree, _serialization).GetDetails(resource);
             model.Methods = Array.Empty<MethodEntry>(); // Reset methods because they can not be invoked on new objects
-
             return model;
         }
 
+        private ActionResult<ResourceModel> Construct(string type, MethodEntry method)
+        {
+            try
+            {
+                var id = _resourceManagement.Create(_resourceTypeTree[type].ResourceType, r => EntryConvert.InvokeMethod(r, method, _serialization));
+                return GetDetails(id);
+            }
+            catch (Exception e)
+            {
+                if (e is ArgumentException or SerializationException or ValidationException)
```

Comment:
Would it be possible to validate the input at the beginning?

Reply:
Unfortunately that is not so easy. The validation logic is nested in the resource management, we would either need to pull that out of it and only use it there and here, or provide a method on the facade. But that is not part of this PR, I just added the catch block here because I stumbled upon the issue by accident. This way you at least get the correct response code 🙈